gips/tiles.py

```diff
@@ -93,20 +93,22 @@ def mosaic(self, datadir, res=None, interpolation=0, crop=False,
                 with utils.error_handler("Error mosaicking " + fout, continuable=True):
                     filenames = [self.tiles[t].filenames[(sensor, product)] for t in self.tiles]
                     images = gippy.GeoImages(filenames)
-                    """
-                    if alltouch and res is None:
-                        timg = gippy.GeoImage(filenames[0])
-                        res = []
-                        res[0] = abs(timg.Affine()[1])
-                        res[1] = abs(timg.Affine()[5])
-                    """
+
                     if self.spatial.site is not None and res is not None:
                         CookieCutter(
                             images, self.spatial.site, str(fout), res[0], res[1],
                             crop, interpolation, {}, alltouch,
                         )
                     else:
-                        mosaic(images, fout, self.spatial.site)
+                        if alltouch is True:
+                            # get the product res, this will indicate that we want "alltouch"
+                            timg = gippy.GeoImage(filenames[0])
+                            affine = images[0].Affine()
+                            product_res = (affine[1], affine[5])
+                        else:
+                            product_res = None
+                        mosaic(images, fout, self.spatial.site, product_res=product_res)
```

Comment:
Optional tidying up, only if you have time:  Set `product_res = None` before the conditional, then drop the else clause.  The conditional can be shortened to `if alltouch:`.  Can you drop the `timg = ` line, as `timg` seems unused?